lib/inspec/resource.rb

```diff
@@ -34,6 +34,11 @@ def self.supports(criteria = nil)
       Inspec::Resource.support_registry[key].push(criteria)
     end
 
+    def resource_id(value = nil)
+      resource_id_value = value || ""
+      @resource_id ||= resource_id_value
```

Comment:
This logic makes the instance variable write-once. That seems like a subtle bug to me.
```
res.resource_id('a')
res.resource_id  => 'a'
res.resource_id('b')
res.resource_id  => 'a'  # this is surprising
```